qemu_v8.mk

```diff
@@ -502,6 +502,22 @@ else
 QEMU_MTE	= off
 endif
 
+QEMU_BASE_ARGS = -nographic
+QEMU_BASE_ARGS += -smp $(QEMU_SMP)
+QEMU_BASE_ARGS += -cpu $(QEMU_CPU)
+QEMU_BASE_ARGS += -d unimp -semihosting-config enable=on,target=native
+QEMU_BASE_ARGS += -m $(QEMU_MEM)
+QEMU_BASE_ARGS += -bios bl1.bin
+QEMU_BASE_ARGS += -initrd rootfs.cpio.gz
+QEMU_BASE_ARGS += -kernel Image
+QEMU_BASE_ARGS += -append 'console=ttyAMA0,38400 keep_bootcon root=/dev/vda2 $(QEMU_KERNEL_BOOTARGS)'
+QEMU_BASE_ARGS += $(QEMU_XEN)
+QEMU_BASE_ARGS += $(QEMU_EXTRA_ARGS)
+QEMU_BASE_ARGS += -machine virt,acpi=off,secure=on,mte=$(QEMU_MTE),gic-version=$(QEMU_GIC_VERSION),virtualization=$(QEMU_VIRT)
+
+QEMU_RUN_ARGS = $(QEMU_BASE_ARGS)
```

Comment:
There's a small problem here. We need the following patch or `make XEN_BOOT=y` fails with an error about `fsdev0` being used twice, and indeed we don't want to add the part related to QEMU_VIRTFS_ENABLE and QEMU_PSS_ENABLE in `make check`, they are useful for `make run` only.
```diff
diff --git a/common.mk b/common.mk
index 1635b6c..4ec7184 100644
--- a/common.mk
+++ b/common.mk
@@ -421,11 +421,11 @@ QEMU_EXTRA_ARGS +=\
 
 ifeq ($(QEMU_VIRTFS_ENABLE),y)
 QEMU_CONFIGURE_PARAMS_COMMON +=  --enable-virtfs
-QEMU_EXTRA_ARGS +=\
+QEMU_RUN_ARGS_COMMON += \
 	-fsdev local,id=fsdev0,path=$(QEMU_VIRTFS_HOST_DIR),security_model=none \
 	-device virtio-9p-device,fsdev=fsdev0,mount_tag=host
 ifeq ($(QEMU_PSS_ENABLE),y)
-QEMU_EXTRA_ARGS +=\
+QEMU_RUN_ARGS_COMMON += \
 	  -fsdev local,id=fsdev1,path=$(QEMU_PSS_HOST_DIR),security_model=mapped-xattr \
 	  -device virtio-9p-device,fsdev=fsdev1,mount_tag=secure
 endif
diff --git a/qemu_v8.mk b/qemu_v8.mk
index 5fa16a7..728f0bc 100644
--- a/qemu_v8.mk
+++ b/qemu_v8.mk
@@ -546,7 +546,7 @@ $(SCMI_DTB): $(SCMI_DTBO) $(QEMU_BUILD)/.stamp_qemu linux arm-tf buildroot
 	cd $(BINARIES_PATH) && fdtoverlay -i qemu_v8.dtb -o $(SCMI_DTB) $(SCMI_DTBO)
 endif
 
-QEMU_RUN_ARGS = $(QEMU_BASE_ARGS)
+QEMU_RUN_ARGS = $(QEMU_BASE_ARGS) $(QEMU_RUN_ARGS_COMMON)
 QEMU_RUN_ARGS += -s -S -serial tcp:127.0.0.1:$(QEMU_NW_PORT) -serial tcp:127.0.0.1:$(QEMU_SW_PORT) 
 
 .PHONY: run-only
```
Please add that change to "qemu_v8: Rework the QEMU arguments for run and check".

Reply:
added it with a small changes to ease integration with SCMI patch